File: data_engineer/pyspark/Football/SourceLoad.py
import json, requests
from pyspark.sql import SparkSession,Row
from pyspark.sql.functions import *
from pyspark.sql import functions as F
from pyspark.sql.types import *
from datetime import *
import os.path as path

spark=SparkSession.builder.master("local[1]")\
    .appName("Music")\
    .config("spark.driver.extraClassPath","/Users/eduardoalberto/opt/spark/jars/mssql-jdbc-7.4.1.jre8.jar" ) \
    .getOrCreate()
spark.sparkContext.setLogLevel("OFF")
                   
# arquivo 
pfile = '/Users/eduardoalberto/LoadFile/BR-Football-Dataset.csv'
nfile = '/Users/eduardoalberto/LoadFile/BR-Football-DatasetOld.csv'

# valida se arquivo exists
def validaArq(df):
    if path.isfile(pfile):
        df = (spark.read
            .format("csv")
            .option("delimiter", ",")
            .option("header", True)
            .option("inferSchema", True)
            .load(pfile))
    else:
        df = (spark.read
            .format("csv")
            .option("delimiter", ",")
            .option("header", True)
            .option("inferSchema", True)
            .load(nfile))
    return df

df = validaArq(pfile)
df.show()
# df.write.mode("overwrite").jdbc("jdbc:sqlserver://localhost:1433;databaseName=DBDWP511", "tb_camp_brasileiro",properties={"user": "sa", "password": "Numsey@Password!"})

File: data_engineer/scala/LoadFile/src/main/scala/SourceLoad.scala
import org.apache.spark.sql._
import org.apache.log4j._
import org.apache.spark.sql.functions._
import java.util.Properties
import java.io.File

object SourceLoad {
  def main(args: Array[String])= {
    Logger.getLogger("org").setLevel(Level.ERROR)

    val spark = SparkSession
      .builder
      .appName("SparkSQL")
      .master("local")
      .getOrCreate()

    val path = "/Users/eduardoalberto/LoadFile/BR-Football-Dataset.csv"
    //valida se arquivo existe
    val arq = new java.io.File(path).exists()
    if(arq){
      val file = path
      val df00 = spark.read.option("delimiter", ",").option("header", "true").csv(file)

    }




  }

}


